### PR TITLE
Drop using tmp files & general script cleanup

### PR DESCRIPTION
**Pull request by jeffbyrnes**
Big win: dropping the creation of a temporary file in `/tmp` in favor of just passing the JSON directly in to cURL.

Also, some general script cleanup, some of which are opinionated. Happy to drop those changes though, if you dislike them.

### CODE REVIEW

**Comment by chadhs** [comment]
thanks @jeffbyrnes ; I'm going to go through the commits and leave some comments; i appreciate the contributions!

**Comment by chadhs** [comment]
https://github.com/chadhs/idonethis-for-alfred/pull/10/commits/43a897a8902ff98f4c3bb8552cbf784897af395d

these variables are changing the internal field separator during the script execution, which is a force of habit when doing this kind of work: http://www.chadstovern.com/changing-your-internal-field-separator/

if everything tested AOK without them we can drop them.

**Comment by chadhs** [comment]
https://github.com/chadhs/idonethis-for-alfred/pull/10/commits/aa99a247a2bf9bfec3dfbdb70c9d2e5af313e71e
i forgot the original reason for using the temp file, but if this is working without issue just using curl and eliminating it that is a big win.

File: src/v2/idt.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ team_id=""
 
 ## static values
 api_version="v2"
-done_date=`date '+%Y-%m-%d'`
+done_date=$(date '+%Y-%m-%d')
```
Comment (chadhs) [inline src/v2/idt.sh:22]: i dig this change, actually my preference as well.

File: src/v2/idt.sh [hunk 2]
```diff
@@ -40,7 +40,7 @@ cat > $tmp_file << DONE
 DONE
 ### submit done to api saving the return response
 post_result=$(curl -s -H "Content-type:application/json" -H "Authorization: Token ${api_token}" --data @${tmp_file} ${base_uri}/entries)
-post_success=`echo ${post_result} | grep created_at`
+post_success=$(echo ${post_result} | grep created_at)
```
Comment (chadhs) [inline src/v2/idt.sh:37]: same here 👍🏼

**Comment by chadhs** [comment]
https://github.com/chadhs/idonethis-for-alfred/pull/10/commits/02b527bb140f7cff43950686c3b236da93a799f1
i'll test these soon and then we'll mark it to be merged!

**Comment by jeffbyrnes** [comment]
@chadhs awesome!

**Comment by jeffbyrnes** [comment]
@chadhs noticed I’d forgotten to remove one more variable, just fixed that up & rebased things.

**Comment by chadhs** [comment]
10-4 @jeffbyrnes we'll get this merged in soon.  ^_^

**Comment by chadhs** [comment]
@jeffbyrnes got busy; testing workflow this morning with 

```
idt ! this is the "big" 'update' with `code`
```

and all works great.

**Pull request merged**